File: 154-find-minimum-in-rotated-sorted-array-ii/154-find-minimum-in-rotated-sorted-array-ii.java
class Solution {
    public int findMin(int[] nums) {
       for(int i = 0; i < nums.length - 1; i++){
           if(nums[i] > nums[i + 1]){
               return nums[i + 1];
           }
        
       }
        if(nums[0] > nums[nums.length - 1]){
            return nums[nums.length - 1];
            
        }
        else{
            return nums[0];
        }
    }
}

File: 154-find-minimum-in-rotated-sorted-array-ii/154-find-minimum-in-rotated-sorted-array-ii.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        
        // int mn=*min_element(nums.begin(),nums.end());
        // return mn;
        int l=0;
        int r=nums.size()-1;
        // return search(nums,l,r);
		
		//Iterative c++ solution
        while(l<r){
            int mid=l+(r-l)/2;
            if(nums[mid]>nums[r]){
                l=mid+1;
            }else if(nums[mid]<nums[r]){
                r=mid;
            }else{
                r=r-1;
            }
        }
        return nums[l];
    
    }
};